nsqd/diskqueue.go

```diff
@@ -341,6 +353,10 @@ func (d *diskQueue) writeOne(data []byte) error {
 	d.writePos += totalBytes
 	atomic.AddInt64(&d.depth, 1)
 
+	// TODO: this is off by one, but we cannot fix this without also fixing
+	// readOne above. We cannot fix readOne above until we guarantee that
+	// every old diskqueue format has been fully processed, otherwise we
+	// will miss one message from old formatted files.
```

Comment:
Sure thing

On Saturday, July 11, 2015, Matt Reiferson <notifications@github.com> wrote:

> In nsqd/diskqueue.go
> <https://github.com/bitly/nsq/pull/600#discussion_r34416476>:
>
> > @@ -341,6 +353,10 @@ func (d *diskQueue) writeOne(data []byte) error {
> >  	d.writePos += totalBytes
> >  	atomic.AddInt64(&d.depth, 1)
> >
> > +	// TODO: this is off by one, but we cannot fix this without also fixing
> > +	// readOne above. We cannot fix readOne above until we guarantee that
> > +	// every old diskqueue format has been fully processed, otherwise we
> > +	// will miss one message from old formatted files.
>
> Fair enough, however for all practical purposes I don't think this is
> something that needs to be addressed other than perhaps the documentation
> around the config param.
>
> Can we remove this comment?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bitly/nsq/pull/600/files#r34416476>.
>